src/Functions/ReplaceStringImpl.h

```diff
@@ -61,7 +61,7 @@ struct ReplaceStringImpl
 
             /// Copy the data without changing
             res_data.resize(res_data.size() + (match - pos));
-            memcpy(&res_data[res_offset], pos, match - pos);
+            memcpySmallAllowReadWriteOverflow15(&res_data[res_offset], pos, match - pos);
```

Comment:
`memcpySmallAllowReadWriteOverflow15` is documented as `Works under assumption, that it's possible to read up to 15 excessive bytes after end of 'src' region and to write any garbage into up to 15 bytes after end of 'dst' region.`. I don't see this prerequisite checked here or anywhere else in this file.